### DISCUSSION TITLE
Funciones deprecadas en las notas 4.01 de variables categóricas

### DISCUSSION THREAD

**Discussion by malalalalala** [rramosp/ai4eng.v1#288] [04 - GESTIÓN DE DATOS]
En las notas 4,01 en la parte de inspeccionar variables categóricas cuando se ejecuta el còdigo aparece que barplot solo recibe un parámetro y que distplot es un método deprecado, qué se puede usar para elaborar esas gráficas entonces?

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[<ipython-input-28-1430da52e5d4>](https://localhost:8080/#) in <cell line: 3>()
     11     plt.subplot(2,4,i+5)
     12     vc = k[c].value_counts()
---> 13     sns.barplot(vc.index, vc.values)
     14     plt.xticks(range(len(vc)), vc.index, rotation="vertical")

TypeError: barplot() takes from 0 to 1 positional arguments but 2 were given

![image](https://github.com/rramosp/ai4eng.v1/assets/59240499/cfd41769-51ec-4612-aa9e-0ecf278ddac3)

**Comment by Felipe-RA** [comment]
Saludos @lauramosdim 

Gracias por el aviso y disculpa la tardanza, parece que la función fue deprecada.

En lo que actualizamos el notebook por favor trata con el sigueinte snippet de código, el cuál genera una representación similar a la original.

```
plt.figure(figsize=(20,8))

# Iterate over your selected columns
for i, c in enumerate(["ExterQual", "HouseStyle", "LandSlope", "Alley"]):  
    # Subplot for distribution plot
    ax1 = plt.subplot(2, 4, i + 1)
    k = d[[c, "SalePrice"]].dropna()
    for v in d[c].dropna().unique():
        sns.histplot(k.SalePrice[k[c] == v], label=v, ax=ax1, element="step", stat="density")  # Using sns.histplot
    plt.title(c)
    plt.yticks([])
    plt.legend()
    
    # Subplot for bar plot
    ax2 = plt.subplot(2, 4, i + 5)
    vc = k[c].value_counts()
    sns.barplot(x=vc.index, y=vc.values, ax=ax2)  # Using keyword arguments x and y
    plt.xticks(rotation="vertical")

plt.tight_layout()  # Adjust layout to avoid overlap
```
